### PR TITLE
refactor: make `color` required & add/update new icons

### PR DESCRIPTION
**Pull request by sgoudham**
This PR makes the `color` key required for all userstyles and adds/updates icons for the [Catppuccin Website (WIP)](https://catppuccin-website.vercel.app/)

Draft PR until I update the documentation

### CODE REVIEW

File: scripts/userstyles.yml [hunk 1]
```diff
@@ -176,14 +179,16 @@ userstyles:
...
+    color: red
     readme:
       app-link: "https://www.instructure.com/canvas"
       current-maintainers: [*thememesniper]
   chatgpt:
     name: ChatGPT
     categories: [artificial_intelligence, productivity]
     icon: openai
-    color: green
+    color: mauve
```
Comment (uncenter) [inline scripts/userstyles.yml:191]: Why the change from green to mauve? Green seems to be more in line with the branding.

File: scripts/userstyles.yml [hunk 2]
```diff
@@ -447,6 +460,8 @@ userstyles:
   mdn:
     name: MDN
     categories: [development, wiki]
+    icon: mdnwebdocs
+    color: text
```
Comment (uncenter) [inline scripts/userstyles.yml:464]: Maybe blue? https://en.wikipedia.org/wiki/MDN_Web_Docs#/media/File:MDN_Web_Docs_logo.svg
Or are we just going off of the Simple Icons colors?

File: scripts/userstyles.yml [hunk 3]
```diff
@@ -469,6 +484,7 @@ userstyles:
   migadu-webmail:
     name: Migadu Webmail
     categories: [email_client, productivity]
+    color: text
```
Comment (uncenter) [inline scripts/userstyles.yml:487]:
```suggestion
    color: blue
```
https://www.migadu.com/svg/logo.svg:

File: scripts/userstyles.yml [hunk 4]
```diff
@@ -516,6 +533,7 @@ userstyles:
   openmediavault:
     name: OpenMediaVault
     categories: [productivity]
+    color: text
```
Comment (uncenter) [inline scripts/userstyles.yml:536]:
```suggestion
    color: sky
```
image

**Review by uncenter — CHANGES_REQUESTED** [review]

File: scripts/userstyles.yml (hunk 1 above)
Comment (sgoudham) [reply scripts/userstyles.yml:191]: image

File: scripts/userstyles.yml (hunk 3 above)
Comment (sgoudham) [reply scripts/userstyles.yml:487]: I'm trying to keep it in line with simple icons

File: scripts/userstyles.yml (hunk 4 above)
Comment (sgoudham) [reply scripts/userstyles.yml:536]: Open media vault doesn't exist in simple icons so just defaulting to text

File: scripts/userstyles.yml (hunk 2 above)
Comment (sgoudham) [reply scripts/userstyles.yml:464]: See below comments

File: scripts/userstyles.yml (hunk 3 above)
Comment (uncenter) [reply scripts/userstyles.yml:487]: It will be blue soon: https://github.com/simple-icons/simple-icons/issues/10498

File: scripts/userstyles.yml (hunk 4 above)
Comment (uncenter) [reply scripts/userstyles.yml:536]: There should ideally still be a color that accurately reflects it right?

File: scripts/userstyles.yml (hunk 3 above)
Comment (sgoudham) [reply scripts/userstyles.yml:487]: We can raise a PR after its updated :+1:

File: scripts/userstyles.yml (hunk 4 above)
Comment (sgoudham) [reply scripts/userstyles.yml:536]: Talked a little in the discord: 

A fair argument can be made about setting it to the brand identity, ultimately my reasoning is that the `color` doesn't matter without the `icon`, and setting it to `text` would give us more consistency on the website. Another argument made was that it can be handled in the code and not the schema but I'd like the the yaml to be a singular source of truth without needing any if checks on the code side.

We may very well revisit this in the future and change it if it severely impacts a future design decision but I, personally, don't think it will.

Comment (uncenter) [reply scripts/userstyles.yml:536]: > Talked a little in the discord:

Apparently I said something wrong! Getting muted for having a discussion does not seem to be fostering the kind of culture I would've hoped/thought you wanted for Catppuccin. So far I've received practically censorship anytime I dare to question the choices of staff/core team...

File: scripts/userstyles.yml (hunk 1 above)
Comment (uncenter) [reply scripts/userstyles.yml:191]: This is ChatGPT, a product of OpenAI, rather than the company itself though.

File: scripts/userstyles.yml (hunk 4 above)
Comment (andreasgrafen) [reply scripts/userstyles.yml:536]: @uncenter

> Getting muted for having a discussion […]
> So far I've received practically censorship anytime I dare to question the choices of staff/core team...

The issue is not that you're engaging in a discussion nor that you are giving feedback and/or raising your concerns.

The underlying issue is more in the way this happens. You seem to be very firm in your opinion and stance on things – which often is a good thing – but in this case can often lead to the impression that you're not actually interested in discussing, but just pushing through your point while disregarding any of the concern others may have. That in return can then get very quickly get tiring for others.

Since Catppuccin is a collaborative project it's important to hear our everyone and value their feedback and response – as long as it's in within a reasonable spectrum – and take that into consideration as well.
But in the end it's often not going to be possible to make everyone happy simply because of the sheer amount of people involved. For those cases there has to be an institution that can make decisions that are weighted more than others. This is where the staff teams come into play. Sure, the reasoning might not always be perfect but the team knows what they're doing. So often those decisions are what immediately is best for the immediate progress of the given project – that said: Nothing is final and there are always improvements that can be made later down the line.

Comment (sgoudham) [reply scripts/userstyles.yml:536]: Adding onto the above, I muted because it seemed like it was going the same way as previous discussions where it was just going around in circles.

Comment (uncenter) [reply scripts/userstyles.yml]: > The underlying issue is more in the way this happens. You seem to be very firm in your opinion and stance on things – which often is a good thing – but in this case can often lead to the impression that you're not actually interested in discussing, but just pushing through your point while disregarding any of the concern others may have. That in return can then get very quickly get tiring for others.

Whether you like it nor not I could say *exactly the same for you*. You and hammy seem to be very firm in your opinion and stance on things which leads to the impression that you're not actually interested in discussing, and just pushing through your point while disregarding any of the concern others may have. This is very tiring for me.

> Since Catppuccin is a collaborative project it's important to hear our everyone and value their feedback and response – as long as it's in within a reasonable spectrum – and take that into consideration as well. But in the end it's often not going to be possible to make everyone happy simply because of the sheer amount of people involved. For those cases there has to be an institution that can make decisions that are weighted more than others. This is where the staff teams come into play. Sure, the reasoning might not always be perfect but the team knows what they're doing. So often those decisions are what is best for the immediate progress of the given project – that said: Nothing is final and there are always improvements that can be made later down the line.

I rephrased and asked a different question. I barely got a response before I was muted. It is quite clear you don't want community feedback in these sorts of matters.

Comment (andreasgrafen) [reply scripts/userstyles.yml]: > Whether you like it nor not I could say exactly the same for you.

Feel free to do so. Yes, we are firm in our opinions how certain things should be handled and I hate to play the age card here but a lot of this is based on past experience.

Since a lot of this revolves around the website; let me reiterate the point both Hammy and I made multiple times already: Right now our goal is to get an MVP going which is following our vision for the website – a vision that has been around for roughly two years now. I am very well aware that there are different proposals, but as of rn they don't fit the vision. The MVP is supposed to convey this vision in order to ease the process of further contribution on that project. This is why comments like »*Just use X instead*« or »*My proposal is better and we should use that as a baseline*« are being brushed off. The issue is that even after stating what I just reiterared, the conversation didn't change. None of the decisions are final, but as long as the MVP is in development, there is no restructuring of the whole project. Period. Once that's done – let me repeat myself – we're more than happy to open up the discussion.

The same thing happened last night for the conversation regarding this PR. You made your point, you got an explanation as to why we are doing it this way instead, and you just re-circled to the beginning not really listening to what was said, but just restating your point that it should be different.
Yes, it could be a code decision to set the quote-unquote default color, **but** setting it in the schema is more reliable in the long run. It is important to have the same result everywhere, so if another project were to use the ports.yml and interpret the colours – if we were to do the `no icon then text` check on the website's code – they would end up with a different colour. So it's going to be in the schema instead as our single source of truth.

**Review by isabelroses — APPROVED** [review]
